app/services/provider_interface/reference_status_report_service.rb

```diff
@@ -0,0 +1,43 @@
+module ProviderInterface
+  class ReferenceStatusReportService
+    attr_reader :provider_application_references, :context
+    include Rails.application.routes.url_helpers
+
+    def initialize(provider_application_references)
+      @provider_application_references = provider_application_references
+    end
+
+    def headers
+      I18n.t('provider_interface.reference_status_report_service.headers')
+    end
+
+    def rows
+      provider_application_references.map do |provider_application_reference|
+        {
+          header: full_name(provider_application_reference),
+          values: [
+            feedback_status_label(provider_application_reference.feedback_status),
+            provider_application_reference.name,
+            provider_application_reference.application_choice_id,
+            reference_link(provider_application_reference),
+          ],
+        }
+      end
+    end
+
+    def full_name(reference)
+      "#{reference.first_name} #{reference.last_name}"
+    end
```

Comment:
This is slightly confusing as at first I read this as the referees name, but is actually the candidates name. There is also already have a method on an application form to return the [full name](https://github.com/DFE-Digital/apply-for-teacher-training/blob/2612d981958f5e52d1b7f8f25f94e73995676a67/app/models/application_form.rb#L259), so we could use that instead